client/components/header.jsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+
+export default class Header extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.handleClickGallery = this.handleClickGallery.bind(this);
+    this.handleClickSignUp = this.handleClickSignUp.bind(this);
+    this.handleClickSignIn = this.handleClickSignIn.bind(this);
+  }
+
+  handleClickGallery() {
+    this.props.setView('catalog', {});
+  }
+
+  handleClickSignUp() {
+    this.props.setView('signup', {});
+  }
+
+  handleClickSignIn() {
+    this.props.setView('signin', {});
+  }
+
+  render() {
+    return (
+      <div>
+        <header>
+          <div className="d-flex">
+            <h3 className="mr-auto" onClick={this.handleClickGallery}>Digital Gallery</h3>
+            <i className="fas fa-user-plus p-2 signUp" onClick={this.handleClickSignUp}> Sign Up</i>
```

Comment:
You want to conditionally render these
Don't render the sign up link if the user is signed in
Don't render the sign out buttin if the user isn't signed in
Don't render the sign in link if the user is signed in